server/sonar-server/src/main/java/org/sonar/server/db/migrations/v50/PopulateProjectsUuidColumnsMigration.java

```diff
@@ -86,98 +91,107 @@ public void handleResult(ResultContext context) {
+      migrationContext.updateComponent(component);
+      saveComponent(migrationContext.writeSession, component);
+    }
+    for (Component component : migrationContext.readSession.getMapper(Migration50Mapper.class).selectDisabledNoneDirectComponentChildrenForProjects(
+      migrationContext.project.getId())) {
+      migrationContext.updateComponent(component);
+      saveComponent(migrationContext.writeSession, component);
     }
   }
 
-  private void migrateDisabledComponents(DbSession readSession, DbSession writeSession, Component project, Map<Long, String> uuidByComponentId) {
-    String projectUuid = getOrCreateUuid(project, uuidByComponentId);
-    for (Component component : readSession.getMapper(Migration50Mapper.class).selectDisabledDirectComponentChildrenForProjects(project.getId())) {
-      component.setUuid(getOrCreateUuid(component, uuidByComponentId));
-      component.setProjectUuid(projectUuid);
-      component.setModuleUuidPath("");
-
-      writeSession.getMapper(Migration50Mapper.class).updateComponentUuids(component);
-      counter.getAndIncrement();
+  private void migrateComponentsWithoutUuid(DbSession readSession, DbSession writeSession) {
+    for (Component component : readSession.getMapper(Migration50Mapper.class).selectComponentsWithoutUuid()) {
+      String uuid = Uuids.create();
+      component.setUuid(uuid);
+      component.setProjectUuid(uuid);
+      saveComponent(writeSession, component);
     }
-    for (Component component : readSession.getMapper(Migration50Mapper.class).selectDisabledNoneDirectComponentChildrenForProjects(project.getId())) {
-      component.setUuid(getOrCreateUuid(component, uuidByComponentId));
-      component.setProjectUuid(projectUuid);
-      component.setModuleUuidPath("");
+  }
 
-      writeSession.getMapper(Migration50Mapper.class).updateComponentUuids(component);
-      counter.getAndIncrement();
-    }
+  private void saveComponent(DbSession writeSession, Component component) {
+    writeSession.getMapper(Migration50Mapper.class).updateComponentUuids(component);
+    counter.getAndIncrement();
   }
 
-  private void updateComponent(Component component, Component project, Map<Long, Component> componentsBySnapshotId, Map<Long, String> uuidByComponentId) {
-    String snapshotPath = component.getSnapshotPath();
-    StringBuilder moduleUuidPath = new StringBuilder();
-    Component lastModule = null;
-    if (!Strings.isNullOrEmpty(snapshotPath)) {
-      for (String s : Splitter.on(".").omitEmptyStrings().split(snapshotPath)) {
-        Long snapshotId = Long.valueOf(s);
-        Component currentComponent = componentsBySnapshotId.get(snapshotId);
-        if (currentComponent.getScope().equals(Scopes.PROJECT)) {
-          lastModule = currentComponent;
-          moduleUuidPath.append(currentComponent.getUuid()).append(".");
+  private static class MigrationContext {
```

Comment:
Indeed, sorry for that, but it was too hard for me to read my own code without this Context class !